yaksh/views.py

```diff
@@ -1843,6 +1843,145 @@ def download_assignment_file(request, quiz_id, question_id=None, user_id=None):
 
 @login_required
 @email_verified
+def upload_users(request, course_id):
+    user = request.user
+    ci = RequestContext(request)
+    course = get_object_or_404(Course, pk=course_id)
+    context = {'course': course}
+
+    if not (course.is_teacher(user) or course.is_creator(user)):
+        msg = 'You do not have permissions to this course.'
+        return complete(request, reason=msg)
+    if request.method == 'POST':
+        if 'csv_file' not in request.FILES:
+            context['message'] = "Please upload a CSV file."
+            return my_render_to_response('yaksh/course_detail.html', context,
+                                         context_instance=ci)
+        csv_file = request.FILES['csv_file']
+        is_csv_file, dialect = is_csv(csv_file)
+        if not is_csv_file:
+            context['message'] = "The file uploaded is not a CSV file."
+            return my_render_to_response('yaksh/course_detail.html', context,
+                                         context_instance=ci)
+        required_fields = ['firstname', 'lastname', 'email']
+        try:
+            reader = csv.DictReader(csv_file.read().decode('utf-8').splitlines(),
+                                    dialect=dialect)
+        except TypeError:
+            context['message'] = "Bad CSV file"
+            return my_render_to_response('yaksh/course_detail.html', context,
+                                         context_instance=ci)
+        stripped_fieldnames = [field.strip().lower() for field in reader.fieldnames]
+        for field in required_fields:
+            if field not in stripped_fieldnames:
+                context['message'] = "The CSV file does not contain the required headers"
+                return my_render_to_response('yaksh/course_detail.html', context,
+                                             context_instance=ci)
+        reader.fieldnames = stripped_fieldnames
+        context['upload_details'] = _read_user_csv(reader, course)
+    return my_render_to_response('yaksh/course_detail.html', context,
+                                 context_instance=ci)
+
+
+def _read_user_csv(reader, course):
+    fields = reader.fieldnames
+    upload_details = ["Upload Summary:"]
```

Comment:
Just a heading along with upload details that will be seen to the moderator after upload.

Reply:
I have currently added form on the top of the page, removed from sidebar, toggle thing I am not doing currently because the upload details are returned to the main course detail page. So it will appear as if upload and upload details are on different page.



Exactly, intentions are not to separate out because the button and instructions look one piece.
So, user will definitely will read the instructions. You can once see UI wise, I am uploading the image.
If not good then will add break.